src/AdoNetCore.AseClient/Internal/InternalConnection.cs

```diff
@@ -662,14 +666,20 @@ private string MakeCommand(string commandText, CommandBehavior behavior, bool na
             return result;
         }
 
-        private IToken[] BuildParameterTokens(AseParameterCollection parameters)
+        private IToken[] BuildParameterTokens(AseCommand command)
         {
             var formatItems = new List<FormatItem>();
             var parameterItems = new List<ParametersToken.Parameter>();
 
-            foreach (var parameter in parameters.SendableParameters)
+            foreach (var parameter in command.Parameters.SendableParameters)
             {
-                var formatItem = FormatItem.CreateForParameter(parameter, _environment);
+                var parameterName = parameter.ParameterName ?? command.Parameters.IndexOf(parameter).ToString();
+                if (!command.FormatItems.TryGetValue(parameterName, out FormatItem formatItem))
+                {
+                    formatItem = FormatItem.CreateForParameter(parameter, _environment, command.CommandType);
+                    command.FormatItems.Add(parameterName, formatItem);
```

Comment:
Why are you maintaining the dictionary of FormatItems? Given it's keyed on parameter name you'll only have duplicates if the parameter itself is duplicated. Is that a valid case?

Reply:
Apparently the current implementation for string typed column is length dependence which works fine until you have to chunk them up due to size limitation during insert. If the size last chunk is much smaller than the others, we can potentially ended up returning a different type for that same column which result in error.

Why use dictionary? Well as a quick alternative for the Linq code where you can quickly find column name via a key instead of loop thru the list and check.